### ISSUE TITLE
Tyepscript sample

### ISSUE THREAD

**Issue by Spiralis** [carloslfu/use-machine#13]
I see that the library is written in Typescript, which is great :).

Is it possible to see a usage sample also written in Typescript?

**Comment by Spiralis** [comment]
I in particular had problems with the `React.createContext` call.
In order to make it compile and to preserve typings I had to change it.

From: 
```
const MachineContext = React.createContext()
```

To:
```
const machine = useMachine<MyContext, MyStateSchema, MyEvent>(MyMachineConfig, myOptionsConfig, myContext);
type TMachine = typeof machine;
const MachineContext = React.createContext<TMachine>(machine)
```
But, when doing that, I get an error: 
`Hooks can only be called inside the body of a function component. (https://fb.me/react-invalid-hook-call)`

Which makes sense. Basically, we need to get the typing of the return of the `useMachine` call, in order to pass it to the `React.createContext<>`call. 

So, tried to use the typescript ReturnType, but couldn't figure it out. 

I had to use some of the types that was created in the useMachine method, so as a hack I got it working like this:
From: 
```
const MachineContext = React.createContext()
```

To:
```
import { EventObject, OmniEvent, State } from "xstate"
import { Interpreter } from "xstate/lib/interpreter"

declare type TSendFn<TContext, TEvent extends EventObject> = (
    event: OmniEvent<TEvent>
) => State<TContext, TEvent>

type TMachine = {
    state: State<MatchContext, MatchEvent>
    context: MatchContext
    send: TSendFn<MatchContext, MatchEvent>
    service: Interpreter<MatchContext, MatchStateSchema, MatchEvent>
}

const MachineContext = React.createContext<TMachine>({
    context: {},
    send: {},
    service: {},
    state: {},
} as TMachine)
```

**Comment by Spiralis** [comment]
Now, I got it working, but is there a better way to do it?

**Comment by Spiralis** [comment]
Usage with the patch in place:
```
import { useMachine, TCreateContext } from "use-machine"
// ...
type TMachine = TCreateContext<MyContext, MyStateSchema, MyEvent>
const MachineContext = React.createContext<TMachine>({} as TMachine)

function App() {
  const machine = useMachine<MyContext, MyStateSchema, MyEvent>(MyMachineConfig, MyMachineOptions, myInitialContext)
  return (
    <MachineContext.Provider value={machine}>
      <Match />
    </MachineContext.Provider>
  )
}
```

**Comment by carloslfu** [comment]
I'll leave this open until I publish the TypeScript example using #14.

**Comment by carloslfu** [comment]
Released in [v1.1.0](https://github.com/carloslfu/use-machine/tree/v1.1.0).
